client/components/search/autocompleters/coupons.js

```diff
@@ -0,0 +1,59 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { computeSuggestionMatch } from './utils';
+import { stringifyQuery } from 'lib/nav-utils';
+import { NAMESPACE } from 'store/constants';
+
+/**
+ * A coupon completer.
+ * See https://github.com/WordPress/gutenberg/tree/master/packages/components/src/autocomplete#the-completer-interface
+ *
+ * @type {Completer}
+ */
+export default {
+	name: 'coupons',
+	className: 'woocommerce-search__product-result', // @TODO remove, fix?
```

Comment:
This isn't currently styled, so you could change it to `'woocommerce-search__coupon-result'`, or omit it entirely. It's not a required option.